### PR TITLE
enable metadata passing in the 'result' object stored alongside the authorization sid

### PR DESCRIPTION
**Pull request by KtorZ**
See #164 

I went for a free object rather than storing directly the scopes as it is a bit more flexible. Let me know.

### CODE REVIEW

File: test/interaction/interaction.test.js [hunk 1]
```diff
@@ -305,6 +306,32 @@ describe('resume after interaction', () => {
...
+          remember: true,
+        },
+        meta: {
+          scope: 'openid',
+        },
+      });
+
+      return this.agent.get('/auth/resume')
+        .expect(() => {
+          const authorization = this.getSession().authorizations[config.client.client_id];
```
Comment (panva) [inline test/interaction/interaction.test.js]: can you use metaFor here? This would cover the last line for codecov

Comment (KtorZ) [reply test/interaction/interaction.test.js]: it seems that the `TestAdapter` doesn't return models' instances but raw objects; 

> this.getSession(...).metaFor is not a function

Any ideas ?

Comment (panva) [reply test/interaction/interaction.test.js]: no ideas, just test the metaFor function then.

**Comment by KtorZ** [comment]
I've modified the `getSession` helper in the test suite to be able to get an instance of the Session object when necessary. This isn't performed by default since it would break a dozen of other tests that actually use the result of `getSession` as a plain JavaScript object.

**Pull request merged**